ipatests/test_integration/test_nfs.py

```diff
@@ -357,25 +359,203 @@ def test_nsswitch_backup_restore_sssd(self):
+
+    @pytest.mark.parametrize(
+        "domain_input,expected_success,test_description", [
+            ("  {ipa_domain}  ", True,
+             "domain with whitespace should be trimmed"),
+            ("{ipa_domain_upper}", True, "uppercase should work"),
+            ("invalid..domain", False, "two dots should be rejected"),
+            (".invalid.dom", False, "leading dot should be rejected"),
+            ("invalid with space", False, "space should be rejected"),
+            ("toolong" + "x" * 60 + ".domain", False,
+             "overly long domain should be rejected"),
+            ("client.test", False,
+             "domain should fail discovery with specific messages"),
+        ])
+    def test_automount_domain_option_validation(
+            self, domain_input, expected_success, test_description):
+        """Parametrized test to validate the --domain option behavior
+           in ipa-client-automount with various domain inputs.
+        """
+        client = self.clients[0]
+        ipa_domain = self.master.domain.name
+
+        # Replace placeholders in domain_input
+        domain_to_test = domain_input.format(
+            ipa_domain=ipa_domain,
+            ipa_domain_upper=ipa_domain.upper()
+        )
+
+        if expected_success:
+            # Should succeed
+            client.run_command([
+                'ipa-client-automount', '--domain', domain_to_test,
+                '--location', 'default', '--debug', '-U'
+            ])
+            time.sleep(WAIT_AFTER_INSTALL)
+
+            # Verify configuration if successful
+            sssd_conf = client.get_file_contents(
+                paths.SSSD_CONF).decode()
+            assert "autofs_provider = ipa" in sssd_conf, \
+                "Autofs provider should be set to ipa"
+
+            # Clean up
+            client.run_command(
+                ['ipa-client-automount', '--uninstall', '-U'],
+                raiseonerr=False
+            )
+            time.sleep(WAIT_AFTER_UNINSTALL)
+        else:
+            # Should fail
+            result = client.run_command([
+                'ipa-client-automount', '--domain', domain_to_test,
+                '--debug'
+            ], stdin_text="n", raiseonerr=False)
+            assert (
+                result.returncode != 0
+                or "Autodiscovery did not find LDAP server" in
+                result.stderr_text
+                or "Invalid domain" in result.stderr_text
+            ), f"Should have failed: {test_description}"
+
+    def test_automount_domain_option_overrides_discovery(self):
+        """Test that explicit --domain option overrides automatic discovery."""
+        client = self.clients[0]
+        ipa_domain = self.master.domain.name
+
+        # First install without domain to establish baseline
+        result_auto = client.run_command([
+            'ipa-client-automount', '--location', 'default', '--debug', '-U'
+        ])
```

Comment:
You shouldn't need to explicitly use --location default. It is, well, the default.